WordPress/gutenberg
# Cover: Add aria-label to fixed and repeated image backgrounds

## What?

When a cover block has a fixed or repeated background, it uses a div with the attribute `role="img"`.
The goal of this PR is to solve two issues:

1. The `role="img"` should not be used if the image is presentational (decorative) and has no `alt` text.
2. If the image is not presentational but content, then it needs to have both `role="img"` and an alternative text. Because `alt=""` is not a valid option for a `div` with `role="img"`, `aria-label` is used instead.

Closes https://github.com/WordPress/gutenberg/issues/49902

## Why?

To improve accessibility when the fixed background or repeated backgrounds are used.

## How?


In block.json, the alt block attribute used the `img` as the selector and the `alt` as the attribute, so I had to remove these
from the block.json. I was not able to make this work with the `<image`, the `<div>`, the `alt`, and the `aria-label `without removing it.

Previous:
```
		"alt": {
			"type": "string",
			"source": "attribute",
			"selector": "img",
			"attribute": "alt",
			"default": ""
		},
```
In the PR:
```
		"alt": {
			"type": "string",
			"default": ""
		},
```

## Testing Instructions

1. Add a cover block with an image background.
2. Enable the options: Fixed background and Repeated background.
3.  Confirm that the alt text field is still present in the block settings sidebar. Previously, this field would have been hidden.
4. With the alt text field **empty**: Please view the source and confirm that the div that has background image in the style attribute does not have a `role="img"` and does not have an `aria-label`.  Do the same on the front.
5. Add the alt text and view the source. Confirm that the div has the alt text content inside an aria-label, and the role="image" is present. Do the same on the front.
6. Deactivate Fixed background and Repeated background,  and confirm that the image now uses the `img` tag and the `alt` attribute. Do the same on the front.

### Testing Instructions for Screen readers


1. Add a cover block with an image background.
2. Navigate to the block settings sidebar.
3. In the settings tab,  navigate past the Layout panel to the Media settings panel.
4. Enable the two first checkboxes: Fixed background and Repeat background.
5. Navigate past to two steppers: left and top.
6. In the text area, add an alternative text.
7. To confirm that the alt text is used with the image background, read the content of the post: The alt will not be announced while the block is selected for editing.
8. Save and view the post or page on the front.
9. On the front, the cover block itself is not focusable. Confirm that the alt text is announced while reading the full page content.


## Screenshots or screencast
